azkustoingest/internal/status/status_table_client.go

```diff
@@ -1,67 +1,86 @@
 package status
 
 import (
+	"context"
+	"encoding/json"
 	"github.com/Azure/azure-kusto-go/azkustoingest/internal/resources"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/data/aztables"
 	"github.com/Azure/azure-sdk-for-go/storage"
 	"github.com/google/uuid"
 )
 
 const (
+	// todo: should we set this timeout?
 	defaultTimeoutMsec = 10000
-	fullMetadata       = "application/json;odata=fullmetadata"
+	fullMetadata       = aztables.MetadataFormatFull
 )
 
 // TableClient allows reading and writing to azure tables.
 type TableClient struct {
 	tableURI resources.URI
-	client   storage.Client
-	service  storage.TableServiceClient
-	table    *storage.Table
+	client   *aztables.Client
+
+	service storage.TableServiceClient
+	table   *storage.Table
 }
 
 // NewTableClient Creates an azure table client.
-func NewTableClient(uri resources.URI) (*TableClient, error) {
-	c, err := storage.NewAccountSASClientFromEndpointToken(uri.URL().String(), uri.SAS().Encode())
+func NewTableClient(client policy.Transporter, uri resources.URI) (*TableClient, error) {
+	cred, err := aztables.NewClientWithNoCredential(uri.String(), &aztables.ClientOptions{
```

Comment:
This returns a client, so why name it `cred`?

Reply:
client is already a param name, I renamed it to tableClient now.